Documentation/contributing/development/debugging.rst

```diff
@@ -0,0 +1,320 @@
+                                                                                                                             2
+                                                                                                                             3
+                                                                                                                             4
+                                                                                                                             5
+                                                                                                                             6
+                                                                                                                             323
+                                                                                                                             6188
+                                                                                                                             15194
+                                                                                                                             18892
+                                                                                                                             25379
+                                                                                                                             29200
+                                                                                                                             32255
+                                                                                                                             33831
+                                                                                                                             16777217
+    &LabelSelector{MatchLabels:map[string]string{reserved.none: ,},MatchExpressions:[]LabelSelectorRequirement{},}   1
+
+In this example 16777217 is used by two selectors, one with ``matchPattern: "*"`` and another empty one. This is because of the policy in use:
+
+.. code:: yaml
+
+    apiVersion: cilium.io/v2
+    kind: CiliumNetworkPolicy
+    metadata:
+      name: "tofqdn-dns-visibility"
+    spec:
+      endpointSelector:
+        matchLabels:
+          any:org: alliance
+      egress:
+      - toPorts:
+          - ports:
+             - port: "53"
+               protocol: ANY
+            rules:
+              dns:
+                - matchPattern: "*"
+      - toFQDNs:
+          - matchPattern: "*"
+
+The L7 DNS rule has an implicit L3 allow-all because it defines only L4 and L7
+sections. This is the second selector in the list, and includes all possible L3
+identities known in the system. In contrast, the first selector, which
+corresponds to the ``toFQDNS: matchName: "*"`` rule would list all identities
+for IPs that came from the DNS Proxy. Other CIDR identities would not be
+included.
+
+*Datapath Plumbing*
```

Comment:
Does Sphinx render this as a paragraph title? I'd expect it to joint it with the next line.